library/src/main/java/com/novoda/downloadmanager/lib/DownloadInfo.java

```diff
@@ -40,10 +40,27 @@ public Reader(ContentResolver resolver, Cursor cursor) {
             mCursor = cursor;
         }
 
-        public DownloadInfo newDownloadInfo(Context context, SystemFacade systemFacade, StorageManager storageManager, DownloadNotifier notifier) {
-            final DownloadInfo info = new DownloadInfo(context, systemFacade, storageManager, notifier);
+        public DownloadInfo newDownloadInfo(
+                Context context,
+                SystemFacade systemFacade,
+                StorageManager storageManager,
+                DownloadNotifier notifier,
+                DownloadClientReadyChecker downloadClientReadyChecker) {
+
+            Fuzz fuzz = new Fuzz();
```

Comment:
Yeah I had the same question.

If you look down the code at https://github.com/novoda/download-manager/pull/29/files#diff-68db38a7048316539cb0931793be1808L242 you will see that is a random value used in order to avoid clashing for a restart time queries.

It is needed to be injected because otherwise when the test calls https://github.com/novoda/download-manager/blob/master/library/src/main/java/com/novoda/downloadmanager/lib/Helpers.java#L39, the code fails